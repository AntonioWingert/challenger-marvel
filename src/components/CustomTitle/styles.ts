import { styled } from "styled-components";
import { CustomTitleProps } from ".";

type TitleProps = Omit<CustomTitleProps, "title">;

export const Container = styled.h1<TitleProps>`
  font-size: ${({ $size, theme }) => $size  && theme["font-sizes"][$size]};
  color: ${({$color, theme}) => $color && theme.colors[$color]};
  font-weight: ${({ $weight, theme }) => $weight && theme.weight[$weight]};
  padding: ${({theme}) => theme.padding.xl};


  @media (max-width: 320px) {
    font-size: ${({ theme }) => theme["font-sizes"]["sm"]};
  }
`;