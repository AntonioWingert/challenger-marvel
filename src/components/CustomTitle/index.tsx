import * as S from './styles';

export type CustomTitleProps = {
  title: string;
  size: 'sm' | 'md' | 'lg' | 'xl' | 'xxl';
  as?: 'h1' | 'h2' | 'h3' | 'h4' | 'h5' | 'h6';
  weight?: 'light' | 'regular' | 'bold';
  color?: 'primary' | 'secondary';
}

export default function CustomTitle({ title = 'Header', size, weight = 'regular', as = "h1", color = "primary" }: CustomTitleProps) {
  return (
    <S.Container size={size} weight={weight} as={as} color={color}>{title}</S.Container>
  );
}