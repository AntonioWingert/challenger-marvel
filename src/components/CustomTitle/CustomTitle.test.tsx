import { screen } from '@testing-library/react';
import render from '../../utils/renderWithRouter';
import CustomTitle from '.';

describe('<CustomTitle />', () => {
  it('should render correctly with default props', () => {
    const {container} = render(
      <CustomTitle title="Header" $size="sm" />
    );

    expect(screen.getByText(/header/i)).toBeVisible();
    expect(screen.getByText(/header/i)).toHaveStyleRule('font-size', '1.6rem');
    expect(container.firstChild).toMatchSnapshot();
  });

  it('should render correctly with props', () => {
    const {container} = render(
      <CustomTitle title="Header" $size="xxl" $weight="bold" $color="secondary" />
    );

    expect(screen.getByText(/header/i)).toBeVisible();
    expect(screen.getByText(/header/i)).toHaveStyleRule('font-size', '3rem');
    expect(screen.getByText(/header/i)).toHaveStyleRule('font-weight', '700');
    expect(screen.getByText(/header/i)).toHaveStyleRule('color', '#F5F5F5');
    expect(container.firstChild).toMatchSnapshot();
  }
  );
});