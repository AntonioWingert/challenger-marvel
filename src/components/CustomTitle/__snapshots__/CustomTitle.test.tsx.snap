// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<CustomTitle /> > should render correctly with default props 1`] = `
.c0 {
  font-size: 1.6rem;
  color: #0D0D0D;
  font-weight: 400;
  padding: 32px;
}

@media (max-width: 320px) {
  .c0 {
    font-size: 1.6rem;
  }
}

<h1
  class="c0"
>
  Header
</h1>
`;

exports[`<CustomTitle /> > should render correctly with props 1`] = `
.c0 {
  font-size: 3rem;
  color: #F5F5F5;
  font-weight: 700;
  padding: 32px;
}

@media (max-width: 320px) {
  .c0 {
    font-size: 1.6rem;
  }
}

<h1
  class="c0"
>
  Header
</h1>
`;
